app/api/controllers/validators/resolvers/ResolveEndDate.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.controllers.validators.resolvers
+
+import api.models.errors.{EndDateFormatError, MtdError}
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+
+import java.time.LocalDate
+
+object ResolveEndDate extends Resolver[String, LocalDate] {
```

Comment:
Are these definitely general-purpose resolvers? No problem if so, but just wondering if they're kind of specific to the endpoint being updated?   If they're specific, they can just go under v2 (v3) /controllers/validators/resolvers and be given a name closer to the field they're for.

Reply:
can also extend LocalDateResolving - gimme a shout a bit later, I can talk you through the best way to do this